inc/dbconnection.class.php

```diff
@@ -506,4 +576,69 @@ static function changeCronTaskStatus($enable = true) {
       $cron->update($input);
    }
 
+
+   /**
+    * Set charset to use for DB connection handler.
+    *
+    * @param mysqli $dbh
+    * @param bool   $use_utf8mb4
+    *
+    * @return void
+    *
+    * @since x.x.x
+    */
+   public static function setConnectionCharset(mysqli $dbh, bool $use_utf8mb4): void {
+      $charset = $use_utf8mb4 ? 'utf8mb4' : 'utf8';
+
+      $dbh->set_charset($charset);
+
+      // The mysqli::set_charset function will make COLLATE to be defined to the default one for used charset.
+      // As we are not using the default COLLATE, we have to define it using `SET NAMES` query.
+      switch ($charset) {
+         case 'utf8':
+            // Legacy charset, should be deprecated in next major version.
+            $dbh->query("SET NAMES 'utf8' COLLATE 'utf8_unicode_ci';");
```

Comment:
We could probably add a Toolbox::deprecated

Reply:
Not now, because migration is not mandatory yet. Will have to be deprecated in the second next major version.